src/util/simplify_expr.cpp

```diff
@@ -319,11 +360,30 @@ static simplify_exprt::resultt<> simplify_string_index_of(
 
     const array_exprt &s2_data = s2_data_opt->get();
 
-    it = std::search(
-      std::next(s1_data.operands().begin(), starting_index),
-      s1_data.operands().end(),
-      s2_data.operands().begin(),
-      s2_data.operands().end());
+    if(last_index)
+    {
+      auto it = std::search(
+        std::next(s1_data.operands().rbegin(), starting_index),
```

Comment:
This use of `starting_index` looks wrong, surely it should be applied to `rend` instead? Make sure the case where reverse-search uses a starting-index limit is tested.

Reply:
This was/is right (although yes, I've added a test).